roctracer/PKGBUILD

```diff
@@ -12,24 +12,25 @@ options=(!staticlibs strip)
 _git='https://github.com/ROCm-Developer-Tools/roctracer'
 source=("roctracer-rocm-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
         'add_string_header.patch::https://patch-diff.githubusercontent.com/raw/ROCmSoftwarePlatform/hsa-class/pull/2.patch')
-sha256sums=('7af5326c9ca695642b4265232ec12864a61fd6b6056aa7c4ecd9e19c817f209e'
+sha256sums=('6fa5b771e990f09c242237ab334b9f01039ec7d54ccde993e719c5d6577d1518'
             '35c45b367d917b8ecf5d4d738e7761699b115b25530ab5528c8a6a4a49424199')
+_dirname="$(basename "$_git")-$(basename "${source[0]}" ".tar.gz")"
 
 build() {
   cmake -B build -Wno-dev \
+        -S "$_dirname" \
         -DCMAKE_INSTALL_PREFIX=/opt/rocm \
-        -DHIP_VDI=1 \
-        "$srcdir/roctracer-rocm-$pkgver"
+        -DHIP_VDI=1
   
   cd "$srcdir/roctracer-rocm-$pkgver/test/hsa"
   patch -Np1 -i "$srcdir/add_string_header.patch"
```

Comment:
should we move the patch code into the `prepare()` function as is convention?

Reply:
The file to which the patch is applied is pulled by `cmake` and not present before `build()`.

Reply:
got it